common-rest/src/main/java/org/uniprot/api/rest/output/converter/JsonMessageConverter.java

```diff
@@ -159,22 +147,19 @@ protected void after(MessageConverterContext<T> context, OutputStream outputStre
         if (!context.isEntityOnly()) {
             generator.writeEndArray();
 
-            if (notNullNotEmpty(context.getFailedIds())) {
-                generator.writeFieldName("failedIds");
-                generator.writeStartArray();
-                for (Object matchedField : context.getFailedIds()) {
-                    writeElement(generator, matchedField);
+            if (notNull(context.getIdMappingStatistics())) {
+                IdMappingStatistics idMappingStatistics = context.getIdMappingStatistics();
+                if (notNullNotEmpty(idMappingStatistics.getFailedIds())) {
+                    writeJsonArray(generator, "failedIds", idMappingStatistics.getFailedIds());
+                }
+                if (notNullNotEmpty(idMappingStatistics.getSuggestedIds())) {
+                    writeJsonArray(
+                            generator, "suggestedIds", idMappingStatistics.getSuggestedIds());
                 }
```

Comment:
How will user know the suggestedIds are for which from ids?

Reply:
Based on the Id pattern. Frontend will handle it.

Reply:
Good point, for UniProt website it would be enough, but for programatic users it will be helpful having the pair. I am going to change it.